cookbooks/bcpc/recipes/cluster_local_repository.rb

```diff
@@ -51,18 +51,26 @@
 end
 
 unless node[:fqdn] == get_bootstrap
-  require 'tempfile'
-  bcpc_apt_key_path = Tempfile.new('bootstrap-gpg-key').path
+  ruby_block 'bcpc-apt-key-create' do
+    block do
+      require 'tempfile'
+      node.run_state['bcpc_apt_key_path'] =
+        Tempfile.new('bootstrap-gpg-key').path
```

Comment:
I cherry-picked the more durable fix out of another branch, using a cache file instead of temp.

(Strictly speaking the "delete" should never have been necessary, because the file should have been cleaned up when the ruby process exited, but Chef somehow broke that feature of the library.)